### PR TITLE
First implementation of Orphanage farm

### PR DESCRIPTION
**Pull request by maxux**

### CODE REVIEW

File: tools/tffarmer/main.go [hunk 1]
```diff
@@ -35,7 +35,7 @@ func main() {
 		cli.StringFlag{
 			Name:   "tnodb, u",
 			Usage:  "URL of the TNODB",
-			Value:  "https://tnodb.dev.grid.tf",
+			Value:  "https://bcdb.dev.grid.tf",
```
Comment (zaibon) [inline tools/tffarmer/main.go:38]: you can also rename the flag to `bcdb`

**Comment by zaibon** [comment]
We should also prevent provisiond to deploy when there is no farmer_id defined.

**Comment by maxux** [comment]
Maybe not prevent everything, it's cool to keep `debug` for example ?

**Comment by zaibon** [comment]
> Maybe not prevent everything, it's cool to keep debug for example ?

I don't really see the point. If there is no farmer_id, then the node should not be used for anything. So there wouldn't be anything to debug anyhow.

File: cmds/provisiond/main.go [hunk 2]
```diff
@@ -56,6 +56,14 @@ func main() {
 		log.Fatal().Msgf("fail to connect to message broker server: %v", err)
 	}
 
+	env := environment.Get()
```
Comment (zaibon) [inline cmds/provisiond/main.go]: it looks like you can do this even before creating the redis client at line 54

Comment (zaibon) [inline cmds/provisiond/main.go:54]: it looks like you can do this even before creating the redis client at line 54

**Review by zaibon — CHANGES_REQUESTED** [review]

**Review by zaibon — APPROVED** [review]

**Pull request merged**